src/MainWindow/main_window.cpp

```diff
@@ -1206,3 +1226,12 @@ void MainWindow::onShowStopMessage(bool showStopCompilationMsg) {
   m_askStopCompilation = showStopCompilationMsg;
   m_settings.setValue(SHOW_STOP_COMPILATION_MESSAGE_KEY, m_askStopCompilation);
 }
+
+void MainWindow::onShowLicenses() {
+  auto licensePath = GlobalSession->Context()->DataPath() / LICENSES_DIR;
+  if (!FileUtils::FileExists(licensePath)) return;
+
+  LicenseViewer viewer(licensePath, m_projectInfo.name, this);
+  centerWidget(viewer);
+  viewer.exec();
```

Comment:
Qt documents says:
Note: Avoid using this function; instead, use open(). Unlike exec(), open() is asynchronous, and does not spin an additional event loop. This prevents a series of dangerous bugs from happening (e.g. deleting the dialog's parent while the dialog is open via exec()). When using open() you can connect to the finished() signal of QDialog to be notified when the dialog is closed.

Reply:
but maybe it is fine in this case

Reply:
for some reason neither show() nor open() show the dialog, so I prefer keeping it as is.
About dialog is also shown via exec().

Reply:
@KochynVolodymyr are you still waiting for this to be resolved ?